File: kotlin-study/practice-kotlin/src/main/java/com/example/practicekotlin/singleton2/java/Main.java
package com.example.practicekotlin.singleton2.java;

class Dummy {

    public static final int CONST = 1;
    private Dummy() {
    }

    public static Dummy getInstance() {
        return new Dummy();
    }
}

public class Main {

    public static void main(String[] args) {
        System.out.println(Dummy.CONST);
        System.out.println(Dummy.getInstance());
    }
}

File: kotlin-study/practice-kotlin/src/main/java/com/example/practicekotlin/singleton2/kt/Main.kt
package com.example.practicekotlin.singleton2.kt

class Dummy {

    private constructor()

    companion object {
        const val CONST = 1
        fun getInstance() = Dummy()
    }
}

fun main() {
    println(Dummy.CONST)
    println(Dummy.getInstance())
}